backend/src/models/streakCalculation.ts

```diff
@@ -0,0 +1,41 @@
+import { User } from "./user.js";
+
+export async function calculateStreak(userID: string): Promise<number> {
+  const user = await User.findById(userID);
+
+  const lastAnsweredTimestamp = user?.lastAnsweredTimestamp;
+  if (!lastAnsweredTimestamp) return 0;
+
+  const today = new Date();
+  today.setHours(0, 0, 0, 0);
+
+  const lastLoginDate = new Date(lastAnsweredTimestamp);
+  lastLoginDate.setHours(0, 0, 0, 0);
+
+  const timeDifference = today.getTime() - lastLoginDate.getTime();
+  const daysDifference = Math.floor(timeDifference / (1000 * 60 * 60 * 24));
+
+  // Check if the user answered a question yesterday
+  const yesterday = new Date();
+  yesterday.setDate(yesterday.getDate() - 1);
+  yesterday.setHours(0, 0, 0, 0);
+
+  const answeredYesterday = user?.lastAnsweredTimestamp &&
+    new Date(user.lastAnsweredTimestamp).toDateString() === yesterday.toDateString();
+
+  // Check if the user answered a question today
+  const answeredToday = user?.lastAnsweredTimestamp &&
+    new Date(user.lastAnsweredTimestamp).toDateString() === today.toDateString();
+
+  if (!answeredYesterday && !answeredToday) {
+    // Reset streak if the user didn't answer a question yesterday or today
+    return 0;
+  }
+
+  // Increase streak by 1 if the user answered a question today and yesterday
+  if (answeredYesterday && answeredToday) {
+    return daysDifference + 1;
```

Comment:
This `daysDifference` is the difference between now and the last time they answered a question. 

If someone answered a question on 4/21, 4/22, and 4/23, they should have a streak of 3, but the lastAnsweredTimestamp would be 1 since they last answered today. What do you think about adding a DB entry to the user? Maybe "streakStartTimestamp" and this function would be the one who determines if they are continuing a streak (don't update) or ending a streak (set to 0) or starting a new one (set to 1).